nautobot/dcim/models/device_components.py

```diff
@@ -515,6 +515,98 @@ class BaseInterface(RelationshipModel):
+        if self.parent_interface is not None:
+
+            # An interface cannot be its own parent
+            if self.parent_interface_id == self.pk:
+                raise ValidationError({"parent_interface": "An interface cannot be its own parent."})
+            # A physical interface cannot have a parent interface
+            if hasattr(self, "type") and self.type != InterfaceTypeChoices.TYPE_VIRTUAL:
+                raise ValidationError(
+                    {"parent_interface": "Only virtual interfaces may be assigned to a parent interface."}
+                )
+
+            # A virtual interface cannot be a parent interface
+            if getattr(self.parent_interface, "type", None) == InterfaceTypeChoices.TYPE_VIRTUAL:
+                raise ValidationError(
+                    {"parent_interface": "Virtual interfaces may not be parents of other interfaces."}
+                )
+
+            # An interface's parent must belong to the same device or virtual chassis
+            if self.parent_interface.parent != self.parent:
+                if self.parent.virtual_chassis is None:
+                    raise ValidationError(
+                        {
+                            "parent_interface": f"The selected parent interface ({self.parent_interface}) belongs to a different device "
+                            f"({self.parent_interface.device})."
+                        }
+                    )
+                elif self.parent_interface.device.virtual_chassis != self.parent_interface.virtual_chassis:
+                    raise ValidationError(
+                        {
+                            "parent_interface": f"The selected parent interface ({self.parent_interface}) belongs to {self.parent_interface.device}, which "
+                            f"is not part of virtual chassis {self.device.virtual_chassis}."
+                        }
+                    )
+
+        # Validate untagged VLAN
+        if self.untagged_vlan and self.untagged_vlan.site_id not in [self.parent.site_id, None]:
+            raise ValidationError(
+                {
+                    "untagged_vlan": (
+                        f"The untagged VLAN ({self.untagged_vlan}) must belong to the same site as the interface's parent "
+                        f"device, or it must be global."
+                    )
+                }
+            )
+
+        # Bridge validation
+        if self.bridge is not None:
+
+            # An interface cannot be bridged to itself
+            if self.bridge_id == self.pk:
+                raise ValidationError({"bridge": "An interface cannot be bridged to itself."})
+
+            # A bridged interface can only be assigned to an interface with type = bridge
+            if hasattr(self.bridge, "type") and self.bridge.type != InterfaceTypeChoices.TYPE_BRIDGE:
+                raise ValidationError({"bridge": f"The selected interface ({self.bridge}) is not a bridge interface."})
```

Comment:
I thought based on previous discussion we were *not* requiring the `bridge` interface to be `TYPE_BRIDGE`, as there are use cases for other sorts of bridging?

Reply:
👍

This continued forward the previous commit's validation. While the form had the restriction removed, the validation still enforced it.